contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -424,57 +424,60 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
         }
         uint256 defaultTime = block.timestamp.sub(creditOracle.gracePeriod());
-        uint256 nextInterestRepay = nextInterestRepayTime[pool][borrower];
-        if (nextInterestRepay != 0 && defaultTime >= nextInterestRepay) {
-            _enterDefault(pool, borrower, DefaultReason.InterestOverdue);
-            return;
-        }
-        uint256 _overBorrowLimitTime = overBorrowLimitTime[pool][borrower];
-        if (_overBorrowLimitTime != 0 && defaultTime >= _overBorrowLimitTime) {
-            _enterDefault(pool, borrower, DefaultReason.TimeLimitExceeded);
-            return;
+        ITrueFiPool2[] memory pools = poolFactory.getSupportedPools();
+        for (uint256 i = 0; i < pools.length; i++) {
+            ITrueFiPool2 pool = pools[i];
+            uint256 nextInterestRepay = nextInterestRepayTime[pool][borrower];
+            if (nextInterestRepay != 0 && defaultTime >= nextInterestRepay) {
+                _enterDefault(borrower, DefaultReason.InterestOverdue);
+                return;
+            }
+            uint256 _overBorrowLimitTime = overBorrowLimitTime[pool][borrower];
+            if (_overBorrowLimitTime != 0 && defaultTime >= _overBorrowLimitTime) {
+                _enterDefault(borrower, DefaultReason.TimeLimitExceeded);
+                return;
+            }
         }
         revert("TrueCreditAgency: Borrower has no reason to enter default at this time");
     }
 
-    function _enterDefault(
-        ITrueFiPool2 pool,
-        address borrower,
-        DefaultReason reason
-    ) private {
-        uint256 principal = borrowed[pool][borrower];
-        (uint8 oldScore, uint8 newScore) = _updateCreditScore(pool, borrower);
-        _rebucket(pool, borrower, oldScore, newScore, 0);
+    function _enterDefault(address borrower, DefaultReason reason) private {
+        ITrueFiPool2[] memory pools = poolFactory.getSupportedPools();
+        for (uint256 i = 0; i < pools.length; i++) {
+            ITrueFiPool2 pool = pools[i];
 
-        uint256 _interest = interest(pool, borrower);
-        borrowerTotalPaidInterest[pool][borrower] = borrowerTotalPaidInterest[pool][borrower].add(_interest);
-        poolTotalPaidInterest[pool] = poolTotalPaidInterest[pool].add(_interest);
+            uint256 principal = borrowed[pool][borrower];
+            (uint8 oldScore, uint8 newScore) = _updateCreditScore(pool, borrower);
+            _rebucket(pool, borrower, oldScore, newScore, 0);
 
-        loanFactory.createDebtToken(pool, borrower, principal.add(_interest));
+            uint256 _interest = interest(pool, borrower);
+            borrowerTotalPaidInterest[pool][borrower] = borrowerTotalPaidInterest[pool][borrower].add(_interest);
+            poolTotalPaidInterest[pool] = poolTotalPaidInterest[pool].add(_interest);
 
-        if (totalBorrowed(borrower) == 0) {
-            borrowingMutex.unlock(borrower);
+            loanFactory.createDebtToken(pool, borrower, principal.add(_interest));
```

Comment:
Should we really deploy the tokens if the borrower didn't take anything from the pool?

Reply:
Thanks for catching this! I remember my previous version had a debt == 0 check, but I must have forgotten to add it back in.